target/esp32/ll_cam.c

```diff
@@ -382,13 +382,13 @@ uint8_t ll_cam_get_dma_align(cam_obj_t *cam)
 }
 
 static bool ll_cam_calc_rgb_dma(cam_obj_t *cam){
-    size_t dma_half_buffer_max = 16 * 1024 / cam->dma_bytes_per_item;
+    size_t dma_half_buffer_max = CONFIG_CAMERA_DMA_BUFFER_SIZE_MAX / 2 / cam->dma_bytes_per_item;
     size_t dma_buffer_max = 2 * dma_half_buffer_max;
     size_t node_max = LCD_CAM_DMA_NODE_BUFFER_MAX_SIZE / cam->dma_bytes_per_item;
 
     size_t line_width = cam->width * cam->in_bytes_per_pixel;
     size_t image_size = cam->height * line_width;
-    if (image_size > (2 * 1024 * 1024) || (line_width > dma_half_buffer_max)) {
+    if (image_size > (4 * 1024 * 1024) || (line_width > dma_half_buffer_max)) {
```

Comment:
this should in reality get the size of PSRAM and divide it by the number of frame buffers

Reply:
When your hardware is an 8MB PSRAM, the available space should be 4MB.

Reply:
this should not be hardcoded to size that is incompatible with our 2MB chips. It should not be const at all, but rather equal to `size_of_psram/fb_count` instead of `4 * 1024 * 1024`

Reply:
Or we can delete this condition, because if the size is too large, it will cause an allocation error in the following code